### PR TITLE
3964 - BulkDownload: Metadata

### LINKED ISSUE #3964: BulkDownload: Metadata
Add metadata to excels divided by variable:
- [x] variable name with table anchor prefix (eg 1a_forestArea)
- [x] data download date
- [x] unit shown in table header (eg Area (1000 ha)

### PR DESCRIPTION
**Pull request by sorja**
For csv files divided by variable, we introduce some metadata:
- table anchor followed by variable name
- date of export
- unit as shown in the table

We introduce also:
- `Objects.getInPath`
- `RowController.getRows` with optional arguments (variableName, includeCols and rowType)

### CODE REVIEW

File: src/server/controller/cycleData/getBulkDownload/getContentVariables.ts [hunk 1]
```diff
@@ -87,6 +91,11 @@ export const getContentVariables = async (props: Props & { fileName: string; ent
         return base
       })
 
+      const { dateExported, unit } = await getMetadata({ assessment, cycle, tableName, csvColumn })
+
+      content[0][csvColumn] = dateExported
+      content[1][csvColumn] = unit
```
Comment (sorja) [inline src/server/controller/cycleData/getBulkDownload/getContentVariables.ts:97]: To only display content on the first 3 rows, we add it like this:

CSV First header is extracted from object keys (so we woujld append csvColumn)
Next row we want to show the date exported and on third row we will show the unit

Note: Here we are still working on object type, therefore we alter only two first objects in the returned `content` array

File: src/server/controller/cycleData/getBulkDownload/utils/getMetadata.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+ */
+const getUnitLabelPath = (tableName: string, cycle: Cycle): string[] => {
+  const pathMap: Record<string, string[]> = {
+    growingStockTotal: ['0', 'cols', '0', 'props', 'labels', cycle.uuid],
+    growingStockComposition2025: ['1', 'cols', '0', 'props', 'labels', cycle.uuid],
+    carbonStockSoilDepth: ['0', 'cols', '0', 'props', 'labels', cycle.uuid],
+  }
+
+  // The unit label is found from the second column of the header row by default
+  const defaultPath = ['0', 'cols', '1', 'props', 'labels', cycle.uuid]
```
Comment (sorja) [inline src/server/controller/cycleData/getBulkDownload/utils/getMetadata.ts:31]: Most of the tables had this format for the unit label.
3 tables had different format

File: src/server/controller/cycleData/getBulkDownload/utils/getMetadata.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+/**
+ * Retrieves the row type for a given table name.
+ *
+ * @param {string} tableName - The name of the table.
+ * @returns {RowType} The row type for the given table.
+ */
+const getRowType = (tableName: string): RowType => {
+  const rowTypeMap: Record<string, RowType> = {
+    carbonStockSoilDepth: RowType.data,
```
Comment (sorja) [inline src/server/controller/cycleData/getBulkDownload/utils/getMetadata.ts:43]: `carbonStockSoilDepth` is the only table without headers (at the moment)

File: src/server/repository/assessment/row/getMany.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+type Props = {
+  assessment: Assessment
+  cycle: Cycle
+  tableName: string
+  variableName?: string
+  includeCols?: boolean
+  rowType?: RowType
+}
+
+export const getMany = (props: Props, client: BaseProtocol = DB): Promise<Array<Row>> => {
```
Comment (minotogna) [inline src/server/repository/assessment/row/getMany.ts:17]: Can u use RowRedisRepositiry.getRows ? (Use RowCaches to generate row keys )

File: src/server/controller/cycleData/getBulkDownload/getContentVariables.ts [hunk 5]
```diff
@@ -87,6 +91,11 @@ export const getContentVariables = async (props: Props & { fileName: string; ent
         return base
       })
 
+      const { dateExported, unit } = await getMetadata({ assessment, cycle, tableName, csvColumn })
```
Comment (minotogna) [inline src/server/controller/cycleData/getBulkDownload/getContentVariables.ts:94]: BulkDownload:getContentVariables->Get metadata from existing controller
@sorja we can address this in a separate issue ?

Comment (sorja) [reply src/server/controller/cycleData/getBulkDownload/getContentVariables.ts:94]: Sure, we can address this separately

**Review by minotogna — APPROVED** [review]

**Pull request merged**